src/app/stores/app_store.ts

```diff
@@ -1962,7 +1966,17 @@ export class AppStore extends BaseStore {
               dataExpression.valueType,
               values
             );
-            dataBinding.orderByCategories = deepClone(categories);
+            try {
+              if (dataBinding.orderByExpression) {
+                dataBinding.orderByCategories = this.getCategoriesForOrderByColumn(
+                  dataBinding
+                );
+              } else {
```

Comment:
consider merging this same logic (the else and catch) into a single finally block